src/Communication/BTSerialCommunicationManager.cpp

```diff
@@ -0,0 +1,211 @@
+#include <Communication/BTSerialCommunicationManager.h>
+
+//Adapted from Finally Functional's SerialBT implementation
+
+BTSerialCommunicationManager::BTSerialCommunicationManager(const VRBTSerialConfiguration_t& configuration, std::unique_ptr<IEncodingManager> encodingManager) 
+	: m_btSerialConfiguration(configuration), 
+	m_encodingManager(std::move(encodingManager)), 
+	m_isConnected(false) 
+{
+	//convert the bluetooth device name from settings into wide
+	const char* name = configuration.name.c_str();
+	size_t newsize = strlen(name) + 1;
+	m_wcDeviceName = new WCHAR[newsize];
```

Comment:
does this need to be released after use?

Reply:
wdym?